### DISCUSSION TITLE
Carla Instance Segmentation - Semantic IDs Embedded in B and G Channel Extraction

### DISCUSSION THREAD

**Discussion by Jadams29** [carla-simulator/carla#5253] [Q&A]
I have found this question asked multiple times but I have not found an answer. I am working with the instance segmentation sensor in carla. In the release summary linked below it states that the semantic ID's are embedded in the G and B channels. How are those ID's extracted from those channels? 

https://carla.org/2021/11/16/release-0.9.13/#instance-segmentation

**Comment by Emcyz** [comment]
unique_id == G + ( B << 8 ) + 0x10000 (if unique_id is larger than 0x10000)
BTW unique_ id is Unreal Engine's id, not carla's id

https://github.com/carla-simulator/carla/discussions/5047

**Comment by dk-kling** [comment]
Is there any way to get the carla's ID with images using a kind of camera sensor?

**Comment by alexkutsan** [comment]
I am also interested in getting Carlas object id to get object details for ground truth.
